emscripten/static/index.js

```diff
@@ -12,12 +12,46 @@
     if (!window.SharedArrayBuffer)
     {
         document.getElementById("loadingWebassembly").innerText = "Error! SharedArrayBuffer is not defined. This page required the CORP and COEP response headers.";
+        return;
     }
     if (!window.WebAssembly)
     {
         document.getElementById("loadingWebassembly").innerText = "Error! This page requires WebAssembly. Please upgrade your browser or enable WebAssembly support.";
+        return;
     }
 
+    let assets;
+    try
+    {
+        let req = await fetch("openrct2.zip");
+        if (!req.ok) {
+            throw new Error("Response is not ok!")
+        }
+        let data = await req.blob();
+        let zip = new JSZip();
+        let contents = await zip.loadAsync(data);
+        assets = {
+            js: URL.createObjectURL(new Blob([await zip.file("openrct2.js").async("uint8array")], {type: 'application/json'})),
+            wasm: URL.createObjectURL(new Blob([await zip.file("openrct2.wasm").async("uint8array")], {type: 'application/wasm'}))
+        }
+    }
+    catch(e)
+    {
+        assets = null;
+        console.warn("Failed to fetch openrct2.zip. Will pull not-compressed files", e);
+    }
+
+    await new Promise(resolve => {
+        const script = document.createElement("script");
+        script.src = assets === null ? "openrct2.js" : assets.js;
```

Comment:
Shouldn’t this be

```suggestion
        script.src = assets === null ? "openrct2.js" : "assets.js";
```

?

Reply:
No it shouldnt, since `assets` is a JSON object. URLs are created just above after decompressing the zip file: https://github.com/OpenRCT2/OpenRCT2/pull/24135/files#diff-93b7ad57d8749eaf006824ceb1fff54331dceee16935ce1a26ed5a4d94d87403R33-R36

I'm open to suggestions on restructuring this part, if its confusing at all